### ISSUE TITLE
Support for @TestSubject to do constructor injection

### ISSUE THREAD

**Issue by sazzer** [easymock/easymock#185]
Sometimes we have classes under test that use Constructor injection instead of Property injection. It would be very useful to be able to use the @Mock/@TestSubject pattern with these objects and have EasyMock automatically instantiate the object and inject mocks into the constructor as appropriate.

Currently the only way that we have of doing this is to do the construction manually in an @Before method, which feels like something that EasyMock can very easily do for us.

**Comment by henri-tremblay** [comment]
Yes. It is indeed a really useful feature to have. Will be in the next release.

**Comment by stevegomez17** [comment]
agree

**Comment by CharanyaR** [comment]
@henri-tremblay I will take up this issue.

**Comment by henri-tremblay** [comment]
@CharanyaR If you can, it will be awesome

**Comment by yoeris** [comment]
Any news on progress for this issue ?
Having the same needs here :-)

**Comment by henri-tremblay** [comment]
Sorry. Not yet. But feel free to contribute ;-)

**Comment by Jidehem** [comment]
There is another workaround without using @Before/@BeforeEach: you can create the mock manually with `createMock(MyClass.class)` and pass it to the constructor of the TestSubject. However, mixing constructor and field injection will not work since the manually created mock will not be registered for injection (only @Mock annotated fields are).
I tried annotating the manually initialized field with @Mock, but it end up with 2 instances of the mock. The logic in `org.easymock.internal.Injector.createMocksForAnnotations(Class<?>, Object, InjectionPlan)` will totally ignore the fact that the field is already set to a mock and replace it with a second instance of the mock.
A possible improvement would simply be to add in this injection method a detection of the current @Mock field value.
If the field is null, then initialize it (current behavior).
If the field is not null, then use it (it does not really matter if the value is a mock or not IMHO) and register it for injection in other `TestSubject`s.

@henri-tremblay : what do you think about this suggestion? I could do this work if it seems relevant.
